src/rebar3_hex.erl

```diff
@@ -15,6 +15,8 @@
 -export_type([task/0]).
 
 init(State) ->
+    filelib:is_dir("_checkouts") andalso
+        rebar_log:log(warn, "_checkouts directory found; this might interfere with publishing", []),
```

Comment:
The provider is initialized at rebar3 load time. This will warn on every single execution of rebar3, including `rebar3 help`.

Reply:
Oh, hahaha. Where should it reside better, then?

Reply:
Should it be per `do`?

Reply:
This should happen in `rebar3_hex_publish`, but we might want to make it an opt in. I would need more information on the case though. It seems like we're wanting to warn if a `_checkouts` dir simply exist in the project you're publishing for. Is the concern that you might accidentally publish the `_checkouts` dir as an artifact?

Reply:
> This should happen in rebar3_hex_publish

I thought about this, but then there's `cut` too, right? So I tried to make it "global" so as to not have to list all the options and forgot to add the warning to some new one that might come later. But I might have gone too far on that thinking...

Reply:
> we might want to make it an opt in

For the warning or the exit? What would the default be, in that case? I think it's reasonable to always get the warning, and only potentially publish...

Reply:
> Is the concern that you might accidentally publish the _checkouts dir as an artifact?

This is probably better expressed in #68.

Reply:
> Maybe a good way to tackle this is make it part of known_exclude_file/2 in rebar3_hex_publish.

I went with Fred's initial suggestion on warning, but can experiment with yours too. Lemme do some stuff here.

Reply:
(rethinking this, I'd go with the warning, since the problem seems to be with `_checkouts`-deps not being locked, which is a `rebar3` feature, and not publishing `_checkouts` as @starbelly thought it might be)

Reply:
As FYI `cut` is going away in v7.

Reply:
Sure. Fun stuff.

Reply:
This is also especially nice since cutting a patch was tagging versions with a `v` prefix, which I don't use 😄